### PR TITLE
fix training loss sequence

### PR DESCRIPTION
**Pull request by AnirudhDagar**
*Description of changes:*
Fix training loss and also refactor code.

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-en/PR-1047/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1047/

**Comment by mli** [comment]
Job d2l-en/PR-1047/2 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1047/

File: chapter_recurrent-neural-networks/sequence.md [hunk 1]
```diff
@@ -177,11 +178,11 @@ def train_net(net, train_iter, loss, epochs, lr):
     for epoch in range(1, epochs + 1):
         for X, y in train_iter:
             trainer.zero_grad()
-            l = loss(net(X), y.reshape(-1, 1))
+            l = loss(net(X), y)
             l.backward()
             trainer.step()
-        print('epoch %d, loss: %f' % (
-            epoch, d2l.evaluate_loss(net, train_iter, loss)))
+        print(f'epoch {epoch},\
```
Comment (astonzhang) [inline chapter_recurrent-neural-networks/sequence.md:184]: multiline f-string style:
```
x = 100
print(f'a {x:d} and '
          f'b {x:d}')
```

File: chapter_recurrent-neural-networks/sequence.md [hunk 2]
```diff
@@ -125,13 +125,14 @@ loss = gluon.loss.L2Loss()
...
-features = torch.zeros((T-tau, tau))
+features = d2l.zeros((T-tau, tau))
 for i in range(tau):
     features[:, i] = x[i: T-tau+i]
 labels = x[tau:]
 
 batch_size, n_train = 16, 600
-train_iter = d2l.load_array((features[:n_train], labels[:n_train]),
+train_iter = d2l.load_array((features[:n_train], 
+                             labels[:n_train].reshape(-1,1)),
```
Comment (astonzhang) [inline chapter_recurrent-neural-networks/sequence.md:135]: add space after ,

File: chapter_recurrent-neural-networks/sequence.md [hunk 3]
```diff
@@ -125,13 +125,14 @@ loss = gluon.loss.L2Loss()
...
```
Comment (astonzhang) [inline chapter_recurrent-neural-networks/sequence.md:135]: add a space after comma

Comment (astonzhang) [reply chapter_recurrent-neural-networks/sequence.md:135]: oh, i mean (-1,1) -> (-1, 1)

Comment (astonzhang) [reply chapter_recurrent-neural-networks/sequence.md:135]: and please remove any trailing space

**Comment by mli** [comment]
Job d2l-en/PR-1047/4 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1047/

**Comment by mli** [comment]
Job d2l-en/PR-1047/3 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1047/
